client/state/data-layer/wpcom/sites/rewind/downloads/index.js

```diff
@@ -66,22 +70,112 @@ const fromApi = data => ( {
 	downloadCount: +data.downloadCount,
 	validUntil: data.validUntil,
 	url: data.url,
+	error: data.error,
 } );
 
-export const updateProgress = ( { dispatch }, { siteId, downloadId }, data ) =>
-	dispatch( updateRewindBackupProgress( siteId, downloadId, data ) );
+/**
+ * When requesting the status of a backup creation, it could be done by a specific download Id or without it.
+ * In the second case a collection is returned so we get the first and latest backup creation object.
+ * If an error property was found, it will display an error card stating it.
+ * Otherwise the backup creation progress will be updated.
+ *
+ * @param {function} dispatch Method to trigger a state change.
+ * @param {number}   siteId   Id of the site for the one we're creating a backup.
+ * @param {object}   apiData  Data returned by a successful response.
+ */
+export const updateProgress = ( { dispatch }, { siteId }, apiData ) => {
+	if ( ! isEmpty( apiData[ 0 ] ) ) {
```

Comment:
Could we add some kind of expectation of `apiData` in a comment here? The use of `apiData[ 0 ]` is confusing because it's oddly-specific and yet unexplained.

Sometimes I find an example in the JSDoc sufficient
Sometimes I like to have a schema there
Sometimes array destructuring clarifies

Basically when I read this I'm asking
 - why is the response an array?
 - what is special about the first item?
 - why are we ignoring all the others?

Reply:
Ok updated adding a comment in JSDoc and destructuring to a meaningful var name.